spark/src/main/scala/io/delta/sharing/spark/DeltaSharingClient.scala

```diff
@@ -215,6 +218,33 @@ private[spark] class DeltaSharingRestClient(
     DeltaTableFiles(version, protocol, metadata, files)
   }
 
+  override def getFiles(table: Table, startingVersion: Long): DeltaTableFiles = {
+    val encodedShareName = URLEncoder.encode(table.share, "UTF-8")
+    val encodedSchemaName = URLEncoder.encode(table.schema, "UTF-8")
+    val encodedTableName = URLEncoder.encode(table.name, "UTF-8")
+    val target = getTargetUrl(
+      s"/shares/$encodedShareName/schemas/$encodedSchemaName/tables/$encodedTableName/query")
+    val (version, lines) = getNDJson(
+      target, QueryTableRequest(Nil, None, None, None, Some(startingVersion)))
+    val protocol = JsonUtils.fromJson[SingleAction](lines(0)).protocol
+    checkProtocol(protocol)
+    val metadata = JsonUtils.fromJson[SingleAction](lines(1)).metaData
+    val addFiles = ArrayBuffer[AddFileForCDF]()
```

Comment:
Question: Should we eventually merge AddFile and AddFileForCDF into AddFile?

Reply:
The only difference is that AddFileForCDF requires `version` and `timestamp` field.
To merge into one protocol:
Pro:
- One less protocol, to represent duplicated file type.
Con:
- A breaking change to the protocol, since AddFileForCDF requires `version` and `timestamp` field.
- If we make the two fields optional, which should work. But they are required for cdf query and queryTable with startingVersion.

Worth considering but I feel it's ok to leave them as is.